AppDB/datastore_server.py

```diff
@@ -658,128 +656,38 @@ def get_composite_indexes_rows(entities, composite_indexes):
-          self.get_entity_key(prefix, e.key().path()))
+    self.logger.debug('Inserting {} entities in transaction with hash {}'.
+                      format(len(entities), txn_hash))
+    txn_values = {}
+    txn_keys = []
 
-    self.logger.debug('Inserting {} entities with transaction hash {}'.
-      format(len(entities), txn_hash))
-    row_values = {}
-    row_keys = []
-
-    kind_row_keys = []
-    kind_row_values = {}
-
-    entities = sorted((self.get_table_prefix(x), x) for x in entities)
-
-    for prefix, group in itertools.groupby(entities, lambda x: x[0]):
-      group_rows = tuple(row_generator(group))
-      new_row_keys = [str(ii[0]) for ii in group_rows]
-      row_keys += new_row_keys
-
-      for ii in group_rows:
-        root_key = self.get_root_key_from_entity_key(str(ii[0]))
-        try:
-          txn_id = txn_hash[root_key]
-          row_values[str(ii[0])] = \
-            {dbconstants.APP_ENTITY_SCHEMA[0]:str(ii[1]), #ent
-            dbconstants.APP_ENTITY_SCHEMA[1]:str(txn_id)} #txnid
-        except KeyError, key_error:
-          self.logger.error('Unable to find {} in {}'.
-            format(root_key, txn_hash))
-          raise key_error
-    for prefix, group in itertools.groupby(entities, lambda x: x[0]):
-      kind_group_rows = tuple(kind_row_generator(group))
-      new_kind_keys = [str(ii[0]) for ii in kind_group_rows]
-      kind_row_keys += new_kind_keys
-
-      for ii in kind_group_rows:
-        kind_row_values[str(ii[0])] = \
-          {dbconstants.APP_KIND_SCHEMA[0]:str(ii[1])}
-
-
-    # TODO do these in ||                        
-    self.datastore_batch.batch_put_entity(dbconstants.APP_ENTITY_TABLE, 
-                                          row_keys, 
-                                          dbconstants.APP_ENTITY_SCHEMA, 
-                                          row_values)    
-
-    self.update_journal(row_keys, row_values, txn_hash)
-
-    self.datastore_batch.batch_put_entity(dbconstants.APP_KIND_TABLE,
-                                          kind_row_keys,
-                                          dbconstants.APP_KIND_SCHEMA, 
-                                          kind_row_values) 
+    for entity in entities:
+      namespace = entity.key().name_space()
+      path = entity.key().path()
+      root_key = self.get_root_key(app, namespace, path.element_list())
+      txn = str(txn_hash[root_key]).zfill(ID_KEY_LENGTH)
+      encoded_path = str(self.__encode_index_pb(path))
+      key = self._SEPARATOR.join([app, txn, namespace, encoded_path])
+      txn_keys.append(key)
+      txn_values[key] = {TRANSACTIONS_SCHEMA[0]: TxnActions.PUT,
+                         TRANSACTIONS_SCHEMA[1]: entity.Encode(),
+                         TRANSACTIONS_SCHEMA[2]: ''}
+
+    self.datastore_batch.batch_put_entity(
+      dbconstants.TRANSACTIONS_TABLE,
+      txn_keys,
+      TRANSACTIONS_SCHEMA,
+      txn_values,
+      ttl=self.MAX_TXN_DURATION * 2
```

Comment:
We could make it smaller. It was just some arbitrary padding so that the data in the transaction table would exist at least as long as the maximum allowed transaction duration.